### DISCUSSION TITLE
Static blocks vs Dynamic blocks - what's the difference

### DISCUSSION THREAD

**Discussion by mburridge** [WordPress/developer-blog-content#21] [Topic Ideas]
Article idea

**Comment by bph** [comment]
Let's get a discussion going, before we put it on the task list = issues.

**Comment by bph** [comment]
Here is a [link to the existing documentation on dynamic blocks.](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/creating-dynamic-blocks/)

**Comment by mburridge** [comment]
> Blocks can be static or dynamic. Static blocks contain rendered content and an object of Attributes used to re-render based on changes. Dynamic blocks require server-side data and rendering while the post content is being generated (rendering).

from [here](https://developer.wordpress.org/block-editor/explanations/architecture/key-concepts/#:~:text=Blocks%20can%20be%20static%20or,is%20being%20generated%20(rendering).).

**Comment by mburridge** [comment]
**Resources**
- https://design.oit.ncsu.edu/2019/03/11/choosing-dynamic-blocks-one/
- https://github.com/WordPress/gutenberg/discussions/35918
- https://kinsta.com/blog/dynamic-blocks/

**Definitions**
> A type of block where the content of which may change and cannot be determined at the time of saving a post, instead calculated any time the post is shown on the front of a site. These blocks may save fallback content or no content at all in their JavaScript implementation, instead deferring to a PHP block implementation for runtime rendering.
- https://developer.wordpress.org/block-editor/explanations/glossary/#dynamic-block

> A type of block where the content of which is known at the time of saving a post. A static block will be saved with HTML markup directly in post content.
- https://developer.wordpress.org/block-editor/explanations/glossary/#static-block

**Comment by mburridge** [comment]
[@wordpress/create-block](https://www.npmjs.com/package/@wordpress/create-block) will build a dynamic block with the following option:

`$ npx @wordpress/create-block --variant dynamic`

**Comment by bph** [comment]
This topic was converted to an Issue #20 and Joni Halabi has agreed to compose the post. Please continue commenting on the issue.

**Comment by bph** [comment]
Published: [Static vs. dynamic blocks: What’s the difference?](https://developer.wordpress.org/news/2023/02/static-vs-dynamic-blocks-whats-the-difference/)
